test/unittests/execute_floating_point_test.cpp

```diff
@@ -100,9 +117,69 @@ struct WasmTypeName
     }
 };
 
+template <typename T>
+class execute_floating_point_types : public testing::Test
+{
+};
+
+template <typename T>
+bytes get_binop_code(Instr opcode)
+{
+    constexpr auto opcode_offset =
+        std::is_same_v<T, float> ?
+            0 :
+            static_cast<uint8_t>(Instr::f64_add) - static_cast<uint8_t>(Instr::f32_add);
+
+    /* wat2wasm
+    (func (param f32 f32) (result f32)
+      (f32.add (local.get 0) (local.get 1))
+    )
+    */
+    auto wasm = from_hex("0061736d0100000001070160027d7d017d030201000a0901070020002001920b");
+    constexpr auto template_type = static_cast<uint8_t>(ValType::f32);
+    constexpr auto template_opcode = static_cast<uint8_t>(Instr::f32_add);
+    EXPECT_EQ(std::count(wasm.begin(), wasm.end(), template_type), 3);
+    EXPECT_EQ(std::count(wasm.begin(), wasm.end(), template_opcode), 1);
+    std::replace(
+        wasm.begin(), wasm.end(), template_type, static_cast<uint8_t>(WasmTypeTraits<T>::valtype));
+    std::replace(wasm.begin(), wasm.end(), template_opcode,
+        static_cast<uint8_t>(static_cast<uint8_t>(opcode) + opcode_offset));
+    return wasm;
+}
+
 using FloatingPointTypes = testing::Types<float, double>;
 TYPED_TEST_SUITE(execute_floating_point_types, FloatingPointTypes, WasmTypeName);
 
+TYPED_TEST(execute_floating_point_types, binop_nan_propagation)
+{
+    // Tests NaN propagation in binary instructions (binop).
+    // If all NaN inputs are canonical the result must be the canonical NaN.
+    // Otherwise, the result must be an arithmetic NaN.
+
+    constexpr Instr opcodes[] = {
+        Instr::f32_add,
```

Comment:
Perhaps add a comment, that these will be replaced with 64-bit versions when `TypeParam` is `double`

Reply:
Done.

Reply:
(I meant closer to this array, because now it looks like the tests checks only `f32_add`)

Reply:
Done.